instrumentation/resque/lib/opentelemetry/instrumentation/resque/instrumentation.rb

```diff
@@ -18,6 +18,28 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
           defined?(::Resque)
         end
 
+        ## Supported configuration keys for the install config hash:
+        #
+        # force_flush: when `true`, all completed spans will be synchronously flushed
+        #   at the end of a job's execution (default: `false`). You will likely wish to
+        #   enable this option for job systems that fork worker processes such as Resque.
+        #
+        # span_naming: when `:job_class`, the span names will be set to
+        #   '<job class name> <operation>'. When `:queue`, the span names
+        #   will be set to '<destination / queue name> <operation>'
+        #
+        # propagation_style: controls how the job's execution is traced and related
+        #   to the trace where the job was enqueued. Can be one of:
+        #
+        #   - :link (default) - the job will be executed in a separate trace. The
+        #     initial span of the execution trace will be linked to the span that
+        #     enqueued the job, via a Span Link.
+        #   - :child - the job will be executed in the same logical trace, as a direct
+        #     child of the span that enqueued the job.
+        #   - :none - the job's execution will not be explicitly linked to the span that
+        #     enqueued the job.
```

Comment:
Happy to split this, unrelated, commentary into a secondary, smaller PR.

Reply:
This doc is fantastic - very much appreciated. ❤️

Reply:
Kudos goes to @ahayworth for doing the heavy lifting.